subsys/net/lib/dhcpv6/dhcpv6.c

```diff
@@ -1415,23 +1415,30 @@ static void dhcpv6_enter_state(struct net_if *iface, enum net_dhcpv6_state state
 
 	switch (iface->config.dhcpv6.state) {
 	case NET_DHCPV6_DISABLED:
-		break;
+		return;
 	case NET_DHCPV6_INIT:
-		return dhcpv6_enter_init(iface);
```

Comment:
To me this way of returning "void" is a bit confusing and I have never used it myself. It gives a hint to reader that the function is suppose to return something even if it is not.
We could make this function a bit readable still and use `break;` instead of empty returns here as the switch statement is at the end of the function. So I suggest that we do
```
switch (...) {
    case NET_DHCPV6_DISABLED:
       break;
    case NET_DHCPV6_INIT:
       dhcpv6_enter_init(iface);
       break;
....
}
```